### PR TITLE
updated documentation

### PR DESCRIPTION
**Pull request by ankitdas13**

### CODE REVIEW

File: documents/papernach.md [hunk 1]
```diff
@@ -0,0 +1,708 @@
...
+|-----------------|---------|------------------------------------------------------------------------------|
+| email*          | string | The customer's email address.                                               |
+| contact*        | string  | The customer's phone number.                      |
+| amount*         | integer  | The amount you want to charge your customer. This should be the same as the amount in the order.                        |
+| currency*        | string  | The 3-letter ISO currency code for the payment. Currently, only `INR` is supported. |
+| order_id*        | string  | The unique identifier of the order created. |
+| customer_id*        | string  | The `customer_id` for the customer you want to charge.  |
+| token*        | string  | The `token_id` generated when the customer successfully completes the authorization payment. Different payment instruments for the same customer have different `token_id`.|
+| recurring*        | string  | Determines if recurring payment is enabled or not. Possible values:<br>* `1` - Recurring is enabled.* `0` - Recurring is not enabled.|
+| description*        | string  | A user-entered description for the payment.|
```
Comment (sofiya02) [inline documents/papernach.md:691]: desc and notes are optional

File: documents/registerEmandate.md [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+
+**Parameters:**
+
+| Name            | Type    | Description                                                                  |
+|-----------------|---------|------------------------------------------------------------------------------|
+| amount*   | integer      | The amount to be captured (should be equal to the authorized amount, in paise) |
+| currency*   | string  | The currency of the payment (defaults to INR)  |
+| customerId*   | string      | The id of the customer to be fetched |
+| method*      | string  | Payment method used to make the registration transaction. Possible value is `emandate`.  |
+| receipt      | string  | Your system order reference id.  |
```
Comment (sofiya02) [inline documents/registerEmandate.md:92]: payment_capture is missing

File: documents/registerEmandate.md [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+Razorpay::Order.create(para_attr)
+```
+**Parameters:**
+
+| Name            | Type    | Description                                                                  |
+|-----------------|---------|------------------------------------------------------------------------------|
+| amount*   | integer      | The amount to be captured (should be equal to the authorized amount, in paise) |
+| currency*   | string  | The currency of the payment (defaults to INR)  |
+| receipt      | string  | Your system order reference id.  |
+| notes | object  | A key-value pair  |
```
Comment (sofiya02) [inline documents/registerEmandate.md:186]: payment_capture is missing

File: documents/registerNach.md [hunk 4]
```diff
@@ -0,0 +1,645 @@
...
+**Parameters:**
+
+| Name            | Type    | Description                                                                  |
+|-----------------|---------|------------------------------------------------------------------------------|
+| amount*   | integer      | The amount to be captured (should be equal to the authorized amount, in paise) |
+| currency*   | string  | The currency of the payment (defaults to INR)  |
+| customerId*   | string      | The id of the customer to be fetched |
+| method*      | string  | Payment method used to make the registration transaction. Possible value is `nach`.  |
+| receipt      | string  | Your system order reference id.  |
+| token  | object  | All keys listed [here](https://razorpay.com/docs/api/recurring-payments/paper-nach/auto-debit/#112-create-an-order) are supported |
```
Comment (sofiya02) [inline documents/registerNach.md:93]: auth_type is mandatory .check other [parameters](https://razorpay.com/docs/api/payments/recurring-payments/paper-nach/auto-debit/#111-create-a-customer)

File: documents/registerNach.md [hunk 5]
```diff
@@ -0,0 +1,645 @@
...
+
+## Create an order to charge the customer
+
+```rb
+param_attr = {
+  "amount": 100,
+  "currency": "INR",
+  "receipt": "Receipt No. 1",
+  "notes": {
+    "key1": "value3",
```
Comment (sofiya02) [inline documents/registerNach.md:318]: payment_capture is missing

File: documents/upi.md [hunk 6]
```diff
@@ -0,0 +1,542 @@
...
+Razorpay::Customer.create(para_attr)
+```
+
+**Parameters:**
+
+| Name          | Type        | Description                                 |
+|---------------|-------------|---------------------------------------------|
+| name*          | string      | Name of the customer                        |
+| email        | string      | Email of the customer                       |
+| contact      | string      | Contact number of the customer              |
```
Comment (sofiya02) [inline documents/upi.md:28]: fail_existing is missing

File: documents/upi.md [hunk 7]
```diff
@@ -0,0 +1,542 @@
...
+**Parameters:**
+
+| Name            | Type    | Description                                                                  |
+|-----------------|---------|------------------------------------------------------------------------------|
+| customer          | object | Details of the customer to whom the registration link will be sent.           |
+| type*        | string  | In this case, the value is `link`.                      |
+| currency*        | string  | The 3-letter ISO currency code for the payment. Currently, only `INR` is supported. |
+| amount*         | integer  | The payment amount in the smallest currency sub-unit.                 |
+| description*    | string  | A description that appears on the hosted page. For example, `12:30 p.m. Thali meals (Gaurav Kumar`).                                                             |
+| subscription_registration           | object  | Details of the authorization payment.                      |
```
Comment (sofiya02) [inline documents/upi.md:154]: sms_notify, email_notify, expire_by, receipt are missing

File: documents/upi.md [hunk 8]
```diff
@@ -0,0 +1,542 @@
...
+|-----------------|---------|------------------------------------------------------------------------------|
+| email*          | string | The customer's email address.                                               |
+| contact*        | string  | The customer's phone number.                      |
+| amount*         | integer  | The amount you want to charge your customer. This should be the same as the amount in the order.                        |
+| currency*        | string  | The 3-letter ISO currency code for the payment. Currently, only `INR` is supported. |
+| order_id*        | string  | The unique identifier of the order created. |
+| customer_id*        | string  | The `customer_id` for the customer you want to charge.  |
+| token*        | string  | The `token_id` generated when the customer successfully completes the authorization payment. Different payment instruments for the same customer have different `token_id`.|
+| recurring*        | string  | Determines if recurring payment is enabled or not. Possible values:<br>* `1` - Recurring is enabled.* `0` - Recurring is not enabled.|
+| description*        | string  | A user-entered description for the payment.|
```
Comment (sofiya02) [inline documents/upi.md:524]: description and notes are optional

**Review by sofiya02 — CHANGES_REQUESTED** [review]

File: documents/upi.md (hunk 8 above)
Comment (ankitdas13) [reply documents/upi.md]: updated

File: documents/upi.md (hunk 7 above)
Comment (ankitdas13) [reply documents/upi.md:155]: added these parameters

File: documents/emandate.md [hunk 9]
```diff
@@ -0,0 +1,483 @@
...
+})
+```
+
+**Parameters:**
+
+| Name            | Type    | Description                                                                  |
+|-----------------|---------|------------------------------------------------------------------------------|
+| amount*          | integer | Amount of the order to be paid                                               |
+| currency*        | string  | Currency of the order. Currently only `INR` is supported.                      |
+| method*        | string  | The authorization method. In this case the value will be `emandate`                      |
```
[the remark this answers is not in the document]
Comment (ankitdas13) [reply documents/emandate.md:87]: added

File: documents/emandate.md [hunk 10]
```diff
@@ -0,0 +1,483 @@
...
+|-----------------|---------|------------------------------------------------------------------------------|
+| email*          | string | The customer's email address.                                               |
+| contact*        | string  | The customer's phone number.                      |
+| amount*         | integer  | The amount you want to charge your customer. This should be the same as the amount in the order.                        |
+| currency*        | string  | The 3-letter ISO currency code for the payment. Currently, only `INR` is supported. |
+| order_id*        | string  | The unique identifier of the order created. |
+| customer_id*        | string  | The `customer_id` for the customer you want to charge.  |
+| token*        | string  | The `token_id` generated when the customer successfully completes the authorization payment. Different payment instruments for the same customer have different `token_id`.|
+| recurring*        | string  | Determines if recurring payment is enabled or not. Possible values:<br>* `1` - Recurring is enabled.* `0` - Recurring is not enabled.|
+| description*        | string  | A user-entered description for the payment.|
```
[the remark this answers is not in the document]
Comment (ankitdas13) [reply documents/emandate.md]: removed asterisk

File: documents/registerEmandate.md (hunk 2 above)
Comment (ankitdas13) [reply documents/registerEmandate.md:92]: added

File: documents/papernach.md [hunk 11]
```diff
@@ -0,0 +1,708 @@
...
+| Name            | Type    | Description                                                                  |
+|-----------------|---------|------------------------------------------------------------------------------|
+| amount*          | integer | Amount of the order to be paid                                               |
+| currency*        | string  | Currency of the order. Currently only `INR` is supported.                      |
+| method*        | string  | The authorization method. In this case the value will be `emandate`                      |
+| receipt         | string  | Your system order reference id.                                              |
+| notes           | object  | A key-value pair                                                             |
+| token           | object  | A key-value pair                                                             |
+
+All parameters listed [here](https://razorpay.com/docs/api/route/#create-transfers-from-payments) are supported
```
[the remark this answers is not in the document]
Comment (ankitdas13) [reply documents/papernach.md]: updated url

File: documents/papernach.md (hunk 1 above)
Comment (ankitdas13) [reply documents/papernach.md]: removed asterisk

File: documents/registerEmandate.md (hunk 3 above)
Comment (ankitdas13) [reply documents/registerEmandate.md:188]: added

File: documents/registerNach.md (hunk 4 above)
Comment (ankitdas13) [reply documents/registerNach.md]: added 
https://github.com/razorpay/razorpay-ruby/blob/update_readme/documents/registerNach.md?plain=1#L93

File: documents/registerNach.md (hunk 5 above)
Comment (ankitdas13) [reply documents/registerNach.md:329]: added 
https://github.com/razorpay/razorpay-ruby/blob/update_readme/documents/registerNach.md?plain=1#L326

File: documents/papernach.md [hunk 12]
```diff
@@ -0,0 +1,708 @@
...
+
+**Parameters:**
+| Name            | Type    | Description                                                                  |
+|-----------------|---------|------------------------------------------------------------------------------|
+| amount*          | integer | Amount of the order to be paid                                               |
+| currency*        | string  | Currency of the order. Currently only `INR` is supported.                      |
+| method*        | string  | The authorization method. In this case the value will be `emandate`                      |
+| receipt         | string  | Your system order reference id.                                              |
+| notes           | object  | A key-value pair                                                             |
+| token           | object  | A key-value pair                                                             |
```
[the remark this answers is not in the document]
Comment (ankitdas13) [reply documents/papernach.md]: added customer id and other params
https://github.com/razorpay/razorpay-ruby/blob/update_readme/documents/papernach.md?plain=1#L106

**Review by sofiya02 — COMMENTED** [review]
Readme LGTM

**Pull request merged**